User/apais/azlinux 3.0 arm64 config

allenpais opened the pull request:
arm64: Clean up Azure Linux kernel configuration

This commit streamlines the ARM64 kernel configuration for Azure Linux environments by removing unused or redundant configuration options and updating settings to optimize performance and stability. Changes include:

- Disabling unused platforms
- Disabling unused device drivers specific to other platforms.

These adjustments ensure that the ARM64 kernel is better tailored to the operational characteristics of Azure, enhancing both efficiency and the overall user experience in cloud environments.

Signed-off-by: Allen Pais <apais@linux.microsoft.com>

allenpais committed:
config: azurelinux 3.0 kernel config

Optimize and Converge Kernel Configuration for Enhanced
Performance and Maintainability

This config is a based on:
https://github.com/microsoft/azurelinux/blob/3.0/SPECS/kernel/config

Key Changes:
  - Deprecated outdated configuration options that no longer
    provide benefits in current hardware and software contexts.

  - We are suggesting these optimizations for our WSL/LCOW
    convergence to azurelinux.

Signed-off-by: Allen Pais <apais@linux.microsoft.com>

allenpais committed:
config: additional updates

 As suggested by Tyler:
   Keep CONFIG_KEXEC_FILE enabled.
   Remove unused CONFIG_RD_* configs.

Signed-off-by: Allen Pais <apais@linux.microsoft.com>

allenpais committed:
config: additional config changes

  - Re-enable support for XEN
  - Additional configs required to boot LCOW

Signed-off-by: Allen Pais <apais@linux.microsoft.com>

allenpais committed:
config: additional configs that are need by LCOW to boot

Signed-off-by: Allen Pais <apais@linux.microsoft.com>

Commit:
kernel: config: build in configs required for WSL2 boot

Build in configs that are called and used during the WSL2 boot.

NF/NFT/NETLINK configs are required for when mirrored networking is
enabled.

DXGKRNL creates /dev/dxg device, when vGPU is available. Driver is not
called by applications unless the /dev/dxg already exists.

Signed-off-by: Kelsey Steele <kelseysteele@microsoft.com>

allenpais committed:
Merge pull request #1 from kelsey-steele/user/kms/x86-kernel-config-updates

kernel: config: build in configs required for WSL2 boot

allenpais committed:
Revert "kernel: config: build in configs required for WSL2 boot"

This reverts commit f96b7b199f7ea5b43fabb28dedc167607bc68bf9.

 Revert WSL config changes.

Signed-off-by: Allen Pais <apais@linux.microsoft.com>

allenpais committed:
arm64: early draft(v0.1)

rlmenge reviewed (changes requested):
Looks like this is intended for 3.0. Please change the target branch to 3.0-dev :)

Commit:
enable configs required for NVIDIA Grace platform

        The following configs are required to use NVIDIA Grace platform for MOS:
        CONFIG_DEVICE_PRIVATE: Enables support of device private memory.
        CONFIG_PCIE_EDR: Enables the error disconnect recover support.
        CONFIG_ARM64_64K_PAGES: Required to use NVIDIA GPU driver.
        CONFIG_PREEMPT_DYNAMIC: Allows dynamic preemption tuning using preempt. It allows the kernel to switch between different preemption models at runtime without the need to recompile the kernel.
        CONFIG_DMABUF_HEAPS and CONFIG_DMABUF_HEAPS_SYSTEM: Enables support for DMA-BUF heaps.
        CONFIG_ARM_CORESIGHT_PMU_ARCH_SYSTEM_PMU: It is related ARM CoreSight PMU and system-wide PMU support. CoreSight is a debugging and performance analysis technology for ARM-based systems, and the PMU is a hardware component that monitors and gathers performance-related data.
        CONFIG_SENSORS_ACPI_POWER: Enables power telemetry through hwmon.

Signed-off-by: Hardik Garg <hargar@microsoft.com>

allenpais committed:
Merge pull request #2 from hargar19/user/hargar/azlinux-3.0-arm64-config

enable configs required for NVIDIA Grace platform

tyhicks commented:
@allenpais how about disabling `CONFIG_HWPOISON_INJECT` on arm64 to match what we're doing on x86_64? It is a testing feature that doesn't need to be enabled in production.
